ui/app/utils/path-tree.js

```diff
@@ -0,0 +1,123 @@
+// @ts-check
+
+// eslint-disable-next-line no-unused-vars
+import VariableModel from '../models/variable';
+// eslint-disable-next-line no-unused-vars
+import MutableArray from '@ember/array/mutable';
```

Comment:
`question`:  Are we actually relying on this array being mutable/reacting to updates?

Reply:
We're relying on the **type** specifically; it's the param for our constructor. This allows us to do `<eventualArrName>.pushObject()` etc. without our typechecker saying that `pushObject doesn't exist on type Array`

Reply:
`question`: Where do we call `pushObject`? I just checked the diff and missed it.

`thought` `non-blocking`: Is there any concern that this would accidentally mutate? When invoking `PathTree` would we want to want to pass this in as `new PathTree(variables.removeArrayObserver)` to prevent a re-computation that would accidentally cause a re-render? My days with the `useEffect` hook have left me totally scarred so I'm just curious here.